build_tools/torchscript_e2e_heavydep_tests/train_models.py

```diff
@@ -0,0 +1,206 @@
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Also available under a BSD-style license. See LICENSE.
+
+import torch
+from torch._decomp import get_decompositions
+from functorch import make_fx
+from torch.nn.utils import _stateless
+from transformers import AutoTokenizer, AutoModelForSequenceClassification
+
+from torch_mlir_e2e_test.torchscript.framework import TestUtils
+from torch_mlir_e2e_test.torchscript.registry import register_test_case
+from torch_mlir_e2e_test.torchscript.annotations import annotate_args, export
+
+from torch import fx
+import copy
+
+torch.manual_seed(0)
+
+############################## Utility Functions ##############################
+
+
+def get_input_annotations(inputs: tuple, dynamic: bool) -> list:
+    """Generates the annotation i.e., shape and dtype for the given inputs, required by torch-mlir module."""
+
+    annotations_list = [None]
+    for i in inputs:
+        temp_list = []
+        if dynamic:
+            temp_list.append([-1 for i in range(len(i.shape))])
+        else:
+            temp_list.append(list(i.shape))
+        temp_list.append(i.dtype)
+        temp_list.append(True)
+        annotations_list.append(tuple(temp_list))
+    return annotations_list
+
+
+def change_fx_graph_return_to_tuple(fx_g: fx.GraphModule):
+    for node in fx_g.graph.nodes:
+        if node.op == "output":
+            # output nodes always have one argument
+            node_arg = node.args[0]
+            out_nodes = []
+            if isinstance(node_arg, list):
+                # Don't return NoneType elements.
+                for out_node in node_arg:
+                    if not isinstance(out_node, type(None)):
+                        out_nodes.append(out_node)
+                # If there is a single tensor/element to be returned don't
+                # create a tuple for it.
+                if len(out_nodes) == 1:
+                    node.args = out_nodes
+                else:
+                    node.args = (tuple(out_nodes), )
+    fx_g.graph.lint()
+    fx_g.recompile()
+    return fx_g
+
+
+def generate_graph(model, inputs, training_fn):
+    # TODO: Pass the decomposition_table according to the model/needs.
+    fx_g = make_fx(training_fn,
+                   decomposition_table=get_decompositions([
+                       torch.ops.aten.embedding_dense_backward,
+                       torch.ops.aten.native_layer_norm_backward,
+                       torch.ops.aten.slice_backward,
+                       torch.ops.aten.select_backward
+                   ]))(dict(model.named_parameters()),
+                       dict(model.named_buffers()), inputs)
+    fx_g.graph.set_codegen(torch.fx.graph.CodeGen())
+    fx_g.recompile()
+    fx_g = change_fx_graph_return_to_tuple(fx_g)
+    ts_g = torch.jit.script(fx_g)
+    # TODO: If not saved/load creates some unnecessary functions that
+    # causes problem during mlir-translate.
+    ts_g.save("temp.ts")
```

Comment:
Can you use tempfile.TemporaryFile instead to avoid creating a bunch of garbage on the filesystem?

Reply:
Just using the temp.TemporaryFile  is giving `  File "/home/prashant/torch-mlir/mlir_venv/lib/python3.9/site-packages/torch/jit/_serialization.py", line 164, in load
    cpp_module = torch._C.import_ir_module_from_buffer(  
RuntimeError: Unrecognized data format ` error. Instead using temp.NamedTemporaryFile and using the .name attribute which stores at `/tmp/...` dir.